src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -71,6 +71,20 @@ def list_role_definitions(cmd, name=None, resource_group_name=None, scope=None,
     return _search_role_definitions(cmd.cli_ctx, definitions_client, name, [scope], custom_role_only)
 
 
+def show_role_definition(cmd, scope=None, name=None, role_id=None):
+    if not any((scope, name, role_id)):
+        raise CLIError('Usage error: Provide --scope and --name, or --id')
+    if not role_id and not (name and scope):
+        raise CLIError('Usage error: Provide both --scope and --name')
```

Comment:
`--scope` should be a required argument to align with `az role assignment create` and `az ad sp create-for-rbac`. Also see https://github.com/Azure/azure-cli/issues/30471.